### DISCUSSION TITLE
Field及其相关组件，不做成受控组件是出于什么考虑？

### DISCUSSION THREAD

**Discussion by BugMaker-Huang** [hellof2e/quark-design#284] [Q&A]
如题

**Comment by sanqi-med** [comment]


**Reply by BugMaker-Huang** [reply]
但即使在我显式传递了value时，我仍然可以通过直接在输入框中输入而修改它的值

**Reply by BugMaker-Huang** [reply]
https://semi.design/zh-CN/input/input#%E5%8F%97%E6%8E%A7%E7%BB%84%E4%BB%B6
这是我所理解的受控组件，当我传递了value时，我必须（通过监听onChange然后）手动去修改传递过去的value变量才能使输入框的数据发生变化。

**Reply by BugMaker-Huang** [reply]
https://legacy.reactjs.org/docs/forms.html#controlled-components
这里是受控组件的定义，尽管它出自于react。

**Comment by sanqi-med** [comment]
> https://semi.design/zh-CN/input/input#%E5%8F%97%E6%8E%A7%E7%BB%84%E4%BB%B6 这是我所理解的受控组件，当我传递了value时，我必须（通过监听onChange然后）手动去修改传递过去的value变量才能使输入框的数据发生变化。

你的理解是对的，我们处理下

**Reply by BugMaker-Huang** [reply]
我并不是认为做成受控组件就是好的/是正确的，我是想请教你们设计组件时做出这种选择的原因QAQ，无论如何感谢您的答复🙏🏻

**Reply by sanqi-med** [reply]
我们的组件都是受控的，这个没问题，这个 filed 开发同学可能没注意到这个问题

**Reply by BugMaker-Huang** [reply]
get，不过似乎你们没做成受控的组件不止field一个
